# <center> Proyecto 1 - Ciencia de datos

Integrentes: 

- Karina Valladares, 18005 

- Alexa Bravo, 18831

- José Eduardo López, 181045

In [1]:
#importamos las librerías que vamos a utilizar 

import pandas as pd #para trabajar con la base de datos y el data frame
#from pandas_profiling import * #para generar code book

In [2]:
df = pd.read_csv("data_mineduc.csv", sep = ";") #leeemos el .csv (NOTA: hay que separar por punto y coma porque así lo genera Tableau)

In [3]:
df

CODIGO DISTRITO   DEPARTAMENTO                    MUNICIPIO  \
0      10-01-0001-42   10-001  SUCHITEPEQUEZ                  MAZATENANGO   
1      10-01-0002-42      NaN  SUCHITEPEQUEZ                  MAZATENANGO   
2      10-01-0003-42   10-018  SUCHITEPEQUEZ                  MAZATENANGO   
3      10-01-0004-42   10-019  SUCHITEPEQUEZ                  MAZATENANGO   
4      10-01-0005-42   10-019  SUCHITEPEQUEZ                  MAZATENANGO   
...              ...      ...            ...                          ...   
33361  08-08-2420-41   08-022    TOTONICAPAN  SAN BARTOLO AGUAS CALIENTES   
33362  08-08-2421-43   08-022    TOTONICAPAN  SAN BARTOLO AGUAS CALIENTES   
33363  08-08-2446-45   08-022    TOTONICAPAN  SAN BARTOLO AGUAS CALIENTES   
33364  08-08-2447-45   08-022    TOTONICAPAN  SAN BARTOLO AGUAS CALIENTES   
33365  08-08-2506-41   08-022    TOTONICAPAN  SAN BARTOLO AGUAS CALIENTES   

                                         ESTABLECIMIENTO  \
0                                               EODP NO1   
1            EPDP ANEXA A COLEGIO REPUBLICA DE GUATEMALA   
2                                        COLEGIO FROEBEL   
3      INSTITUTO PRIVADO MIXTO TECNOLOGICO DEL SUROCC...   
4                   COLEGIO PRIVADO MIXTO LICEO MAZATECO   
...                                                  ...   
33361                                  COPB ANEXO A EORM   
33362                                               EORM   
33363  NUCLEO FAMILIAR EDUCATIVO PARA EL DESARROLLO N...   
33364  NUCLEO FAMILIAR EDUCATIVO PARA EL DESARROLLO N...   
33365                                  COPB ANEXO A EORM   

                                               DIRECCION    TELEFONO  \
0      11 CALLE FINAL ZONA 1 FRENTE INST TECNICO INDU...  58359230.0   
1                                 3A AVE 7A CALLE ZONA 1         NaN   
2               11 CALLE FINAL ZONA 3 COLONIA SAN ANDRES  41341472.0   
3                                          FINCA CHOJOJA         NaN   
4                                      6A AVE 221 ZONA 1  40281538.0   
...                                                  ...         ...   
33361                        PARAJE CHUIAJ ALDEA TZANJON  40574929.0   
33362                        PARAJE CHUIAJ ALDEA TZANJON  40574929.0   
33363                CASERIO CHOCACULEU ALDEA PARRAXCHAJ  50495174.0   
33364                                   ALDEA CHOCANULEU  47702945.0   
33365                  PARAJE PACHOC ALDEA TIERRA BLANCA  46293218.0   

                                       SUPERVISOR  \
0                      RITA ELENA GONZALEZ GARCIA   
1                                             NaN   
2                     JUANA MARIA CIFUENTES PEREZ   
3      REYNA DEL ROSARIO CHAVEZ BARROW DE PACHECO   
4      REYNA DEL ROSARIO CHAVEZ BARROW DE PACHECO   
...                                           ...   
33361                JOSÉ FELICIANO PÉREZ CHAMPET   
33362                JOSÉ FELICIANO PÉREZ CHAMPET   
33363                JOSÉ FELICIANO PÉREZ CHAMPET   
33364                JOSÉ FELICIANO PÉREZ CHAMPET   
33365                JOSÉ FELICIANO PÉREZ CHAMPET   

                                       DIRECTOR                 NIVEL  \
0                                           NaN              PARVULOS   
1                                           NaN              PARVULOS   
2      LIGIA ALEJANDRA CIFUENTES GODOY DE LOPEZ              PARVULOS   
3               JOSE FELICIANO ESCOBAR MARTINEZ              PARVULOS   
4                  SANDRY ZIOMARA MANCIO MENDEZ              PARVULOS   
...                                         ...                   ...   
33361                JULIO CRISTOBAL PUAC TUMAX  PREPRIMARIA BILINGUE   
33362                JULIO CRISTOBAL PUAC TUMAX              PRIMARIA   
33363         ZOILA RADARANY CHAMORRO VELÁSQUEZ                BASICO   
33364                JAMES DANIEL SONTAY TORRES                BASICO   
33365    ERICKA ANABELLA XILOJ PEREZ DE CAPRIEL  PREPRIMARIA BI

In [4]:
df.isna().sum().to_frame() #revisamos si hay datos faltantes en nuestras variables

0
CODIGO               0
DISTRITO          1706
DEPARTAMENTO        22
MUNICIPIO           22
ESTABLECIMIENTO     28
DIRECCION          303
TELEFONO          8716
SUPERVISOR        1678
DIRECTOR         12826
NIVEL               22
SECTOR              22
AREA                22
STATUS              22
MODALIDAD           22
JORNADA             22
PLAN                22
DEPARTAMENTAL       22

In [5]:
df.isnull().sum().to_frame() #revisamos si hay datos faltantes en nuestras variables

0
CODIGO               0
DISTRITO          1706
DEPARTAMENTO        22
MUNICIPIO           22
ESTABLECIMIENTO     28
DIRECCION          303
TELEFONO          8716
SUPERVISOR        1678
DIRECTOR         12826
NIVEL               22
SECTOR              22
AREA                22
STATUS              22
MODALIDAD           22
JORNADA             22
PLAN                22
DEPARTAMENTAL       22

### Descripción de los datos

- Cantidad de columnas: 17 (variables)
- Cantidad de filas: 33366 (registros)

### Lista de variables 

- CODIGO: 0 faltantes (no se requiere limpieza).  Es un número asignado por el Ministerio de Educación al establecimiento.

- DISTRITO: 1706 faltantes (complicado de limpiar). Sección asignada, por medio de un código particular.

- DEPARTAMENTO: 22 faltantes (fácil de limpiar). El nombre del departamento al que pertenece la institución.

- MUNICIPIO: 22 faltantes (fácil de limpiar). Nivel municipal.

- ESTABLECIMIENTO: 28 faltantes (fácil de limpiar). Nombre de la institución educativa.

- DIRECCION: 303 faltantes (dificultad media de limpieza). Dirección del establecimiento.

- TELEFONO: 8,716 faltantes (complicado de limpiar). Número telefónico del establecimiento.

- SUPERVISOR: 1,678 faltantes (complicado de limpiar). Trabajador asignado por el Ministerio de Educación según el distrito al que pertenece.

- DIRECTOR: 12,826 faltantes (complicado de limpair). Alto mando en el establecimiento.

- NIVEL: 22 faltantes (fácil de limpiar). Alto nivel de educación impartida.

- SECTOR: 22 faltantes (fácil de limpiar). Si pertenece al sector privado o es público.

- AREA: 22 faltantes (fácil de limpiar). Si está ubicado en un área rural o urbana.

- STATUS: 22 faltantes (fácil de limpiar). Si el establecimiento está abierto, o cerrado.

- MODALIDAD: 22 faltantes (fácil de limpiar). Monolingue, bilingue o poliglota.

- JORNADA: 22 faltantes (fácil de limpiar).  Si es matutina, vespertina, nocturna o cualquier combinación de estas.

- PLAN: 22 faltantes (fácil de limpiar). Si está en un plan diario, fines de semana, sábado o a distancia.

- DEPARTAMENTAL: 22 faltantes (fácil de limpiar). La jurisdicción a la que pertenece el establecimiento.

### Estrategia de limpieza

- Para las variables DEPARTAMENTO, MUNICIPIO, ESTABLECIMIENTO, NIVEL, SECTOR, AREA, 
  STATUS, MODALIDAD, JORNADA, PLAN, DEPARTAMENTAL tienen pocos datos faltantes por lo que podemos simplemente eliminar los registros dado que son relativamente pocos (menos de 300) comparado con la cantidad total de registros (33, 366). 
  
 
 - DIRECCION: revisar errores ortográficos y duplicados.
 
 - SUPERVISOR: intentar limpiar las columnas para solo dejar palabras clave. Revisar errores ortográficas y duplicados.
 
 - DIRECTOR: eliminar la variable dado que hay demasiados datos faltantes y no se considera relevante para un análisis posterior.
 
 - TELEFONO: asegurarse de que quede únicamente un número de teléfono por columna, eliminar los guiones o espacios en blanco. Separar los números de teléfono de forma adecuada.
 
 - AREA: se modifica su categoría a un código numérico
 
 - STATUS: se modifica su categoría a un código numérico
 
 - MODALIDAD: se modifica su categoría a un código numérico
 
 - JORNADA: se modifica su categoría a un código numérico
 
 - SECTOR: se modifica su categoría a un código numérico
 
 - PLAN: se modifica su categoría a un código numérico
 
 

In [6]:
#creamos una función para las categorías, así se les asigna un valor numérico
def categoria_num(df, categoria, valor_0, valor_1, valor_2, valor_3, valor_4, valor_5, valor_6, valor_7, 
                    valor_8):
    resultado = []
    
    for i in range(0, len(df)):
        if df[categoria][i]== valor_0:
            resultado.append(0)
        elif df[categoria][i] == valor_1:
            resultado.append(1)
        elif df[categoria][i] == valor_2:
            resultado.append(2)
        elif df[categoria][i] == valor_3:
            resultado.append(3)
        elif df[categoria][i] == valor_4:
            resultado.append(4)
        elif df[categoria][i] == valor_5:
            resultado.append(5)  
        elif df[categoria][i] == valor_6:
            resultado.append(6)
        elif df[categoria][i] == valor_7:
            resultado.append(7)
        elif df[categoria][i] == valor_8:
            resultado.append(8)
        else:
            resultado.append(df[categoria][i])
    df[categoria] = resultado
    
    return print(valor_0, ": 0\n", 
                 valor_1, ": 1\n",
                valor_2, ": 2\n",
                valor_3, ": 3\n",
                valor_4, ": 4\n",
                valor_5, ": 5\n",
                 valor_6, ": 6\n",
                 valor_7, ": 7\n",
                 valor_8, ": 8\n")

### Convertir las variables categóricas en una clasificación numérica para un manejo más sencillo.

|  Variable |                 |                       |         Código        |                       |                  |      |
|:---------:|:---------------:|:---------------------:|:---------------------:|:---------------------:|:----------------:|:----:|
|           |        0        |           1           |           2           |           3           |         4        |   5+|
|    AREA   |      URBANA     |         RURAL         |    SIN ESPECIFICAR    |           -           |         -        |   -  |
|   STATUS  |     ABIERTA     | CERRADA TEMPORALMENTE | TEMPORAL CONTRATO 021 | TEMPORAL NOMBRAMIENTO | TEMPORAL TITULOS |   -  |
| MODALIDAD |     BILINGUE    |       MONOLINGUE      |           -           |           -           |         -        |   -  |
|  JORNADA  |      DOBLE      |       INTERMEDIA      |        MATUTINA       |        NOCTURNA       |    VESPERTINA    |   -  |
|   SECTOR  |   COOPERATIVA   |       MUNICIPAL       |        OFICIAL        |         RURAL         |         -        |   -  |
|    PLAN   | DIARIO(REGULAR) |        SABATINO       |     FIN DE SEMANA     |        DOMINCAL       |       MIXTO      | OTRO |

#### Limpiamos la variable "AREA"

In [7]:
categoria_num(df, "AREA", "URBANA", "RURAL", "SIN ESPECIFICAR", "", "", "", "", "", "")

URBANA : 0
 RURAL : 1
 SIN ESPECIFICAR : 2
  : 3
  : 4
  : 5
  : 6
  : 7
  : 8



In [8]:
df.groupby("AREA").count()

CODIGO  DISTRITO  DEPARTAMENTO  MUNICIPIO  ESTABLECIMIENTO  DIRECCION  \
AREA                                                                          
0.0    12517     11946         12517      12517            12517      12261   
1.0    20825     19712         20825      20825            20819      20800   
2.0        2         2             2          2                2          2   

      TELEFONO  SUPERVISOR  DIRECTOR  NIVEL  SECTOR  STATUS  MODALIDAD  \
AREA                                                                     
0.0      10919       11974      8924  12517   12517   12517      12517   
1.0      13729       19712     11614  20825   20825   20825      20825   
2.0          2           2         2      2       2       2          2   

      JORNADA   PLAN  DEPARTAMENTAL  
AREA                                 
0.0     12517  12517          12517  
1.0     20825  20825          20825  
2.0         2      2              2

#### Limpiamos la variable "STATUS"

In [9]:
categoria_num(df, "STATUS", "ABIERTA", "CERRADA TEMPORALMENTE", "TEMPORAL CONTRATO 021", "TEMPORAL NOMBRAMIENTO", "TEMPORAL TITULOS", "", "", "", "")

ABIERTA : 0
 CERRADA TEMPORALMENTE : 1
 TEMPORAL CONTRATO 021 : 2
 TEMPORAL NOMBRAMIENTO : 3
 TEMPORAL TITULOS : 4
  : 5
  : 6
  : 7
  : 8



In [10]:
df.groupby("STATUS").count()

CODIGO  DISTRITO  DEPARTAMENTO  MUNICIPIO  ESTABLECIMIENTO  DIRECCION  \
STATUS                                                                          
0.0      27357     27305         27357      27357            27353      27291   
1.0       5740      4110          5740       5740             5738       5535   
2.0          5         5             5          5                5          5   
3.0        109       108           109        109              109        107   
4.0        133       132           133        133              133        125   

        TELEFONO  SUPERVISOR  DIRECTOR  NIVEL  SECTOR   AREA  MODALIDAD  \
STATUS                                                                    
0.0        22197       27305     18487  27357   27357  27357      27357   
1.0         2319        4138      1917   5740    5740   5740       5740   
2.0            0           5         0      5       5      5          5   
3.0           68         108        84    109     109    109        109   
4.0           66         132        52    133     133    133        133   

        JORNADA   PLAN  DEPARTAMENTAL  
STATUS                                 
0.0       27357  27357          27357  
1.0        5740   5740           5740  
2.0           5      5              5  
3.0         109    109            109  
4.0         133    133            133

#### Limpiamos la variable "MODALIDAD"

In [11]:
df.groupby("MODALIDAD").count()

CODIGO  DISTRITO  DEPARTAMENTO  MUNICIPIO  ESTABLECIMIENTO  \
MODALIDAD                                                                
BILINGUE      6724      6398          6724       6724             6723   
MONOLINGUE   26620     25262         26620      26620            26615   

            DIRECCION  TELEFONO  SUPERVISOR  DIRECTOR  NIVEL  SECTOR   AREA  \
MODALIDAD                                                                     
BILINGUE         6706      4694        6398      4027   6724    6724   6724   
MONOLINGUE      26357     19956       25290     16513  26620   26620  26620   

            STATUS  JORNADA   PLAN  DEPARTAMENTAL  
MODALIDAD                                          
BILINGUE      6724     6724   6724           6724  
MONOLINGUE   26620    26620  26620          26620

In [12]:
categoria_num(df, "MODALIDAD", "BILINGUE", "MONOLINGUE", "", "", "", "", "", "", "")

BILINGUE : 0
 MONOLINGUE : 1
  : 2
  : 3
  : 4
  : 5
  : 6
  : 7
  : 8



In [13]:
df.groupby("MODALIDAD").count()

CODIGO  DISTRITO  DEPARTAMENTO  MUNICIPIO  ESTABLECIMIENTO  \
MODALIDAD                                                               
0.0          6724      6398          6724       6724             6723   
1.0         26620     25262         26620      26620            26615   

           DIRECCION  TELEFONO  SUPERVISOR  DIRECTOR  NIVEL  SECTOR   AREA  \
MODALIDAD                                                                    
0.0             6706      4694        6398      4027   6724    6724   6724   
1.0            26357     19956       25290     16513  26620   26620  26620   

           STATUS  JORNADA   PLAN  DEPARTAMENTAL  
MODALIDAD                                         
0.0          6724     6724   6724           6724  
1.0         26620    26620  26620          26620

#### Limpiamos la variable "JORNADA"

In [14]:
df.groupby("JORNADA").count()

CODIGO  DISTRITO  DEPARTAMENTO  MUNICIPIO  ESTABLECIMIENTO  \
JORNADA                                                                  
DOBLE         4250      4190          4250       4250             4249   
INTERMEDIA     107        89           107        107              107   
MATUTINA     22142     20939         22142      22142            22138   
NOCTURNA       498       403           498        498              498   
VESPERTINA    6347      6039          6347       6347             6346   

            DIRECCION  TELEFONO  SUPERVISOR  DIRECTOR  NIVEL  SECTOR   AREA  \
JORNADA                                                                       
DOBLE            4230      3913        4191      3518   4250    4250   4250   
INTERMEDIA        106        71          93        52    107     107    107   
MATUTINA        21976     15316       20953     12367  22142   22142  22142   
NOCTURNA          477       330         407       218    498     498    498   
VESPERTINA       6274      5020        6044      4385   6347    6347   6347   

            STATUS  MODALIDAD   PLAN  DEPARTAMENTAL  
JORNADA                                              
DOBLE         4250       4250   4250           4250  
INTERMEDIA     107        107    107            107  
MATUTINA     22142      22142  22142          22142  
NOCTURNA       498        498    498            498  
VESPERTINA    6347       6347   6347           6347

In [15]:
categoria_num(df, "JORNADA", "DOBLE", "INTERMEDIA","MATUTINA", "NOCTURNA", "VESPERTINA", "", "", "", "")

DOBLE : 0
 INTERMEDIA : 1
 MATUTINA : 2
 NOCTURNA : 3
 VESPERTINA : 4
  : 5
  : 6
  : 7
  : 8



In [16]:
df.groupby("JORNADA").count()

CODIGO  DISTRITO  DEPARTAMENTO  MUNICIPIO  ESTABLECIMIENTO  \
JORNADA                                                               
0.0        4250      4190          4250       4250             4249   
1.0         107        89           107        107              107   
2.0       22142     20939         22142      22142            22138   
3.0         498       403           498        498              498   
4.0        6347      6039          6347       6347             6346   

         DIRECCION  TELEFONO  SUPERVISOR  DIRECTOR  NIVEL  SECTOR   AREA  \
JORNADA                                                                    
0.0           4230      3913        4191      3518   4250    4250   4250   
1.0            106        71          93        52    107     107    107   
2.0          21976     15316       20953     12367  22142   22142  22142   
3.0            477       330         407       218    498     498    498   
4.0           6274      5020        6044      4385   6347    6347   6347   

         STATUS  MODALIDAD   PLAN  DEPARTAMENTAL  
JORNADA                                           
0.0        4250       4250   4250           4250  
1.0         107        107    107            107  
2.0       22142      22142  22142          22142  
3.0         498        498    498            498  
4.0        6347       6347   6347           6347

#### Limpiamos la variable "SECTOR"

In [17]:
df.groupby("SECTOR").count()

CODIGO  DISTRITO  DEPARTAMENTO  MUNICIPIO  ESTABLECIMIENTO  \
SECTOR                                                                    
COOPERATIVA     992       978           992        992              992   
MUNICIPAL       322       303           322        322              322   
OFICIAL       20038     19198         20038      20038            20032   
PRIVADO       11992     11181         11992      11992            11992   

             DIRECCION  TELEFONO  SUPERVISOR  DIRECTOR  NIVEL   AREA  STATUS  \
SECTOR                                                                         
COOPERATIVA        974       848         978       593    992    992     992   
MUNICIPAL          320       204         303       207    322    322     322   
OFICIAL          19944     13362       19197     11190  20038  20038   20038   
PRIVADO          11825     10236       11210      8550  11992  11992   11992   

             MODALIDAD  JORNADA   PLAN  DEPARTAMENTAL  
SECTOR                                                 
COOPERATIVA        992      992    992            992  
MUNICIPAL          322      322    322            322  
OFICIAL          20038    20038  20038          20038  
PRIVADO          11992    11992  11992          11992

In [18]:
categoria_num(df, "SECTOR", "COOPERATIVA", "MUNICIPAL","OFICIAL", "PRIVADO", "", "", "", "", "")

COOPERATIVA : 0
 MUNICIPAL : 1
 OFICIAL : 2
 PRIVADO : 3
  : 4
  : 5
  : 6
  : 7
  : 8



In [19]:
df.groupby("SECTOR").count()

CODIGO  DISTRITO  DEPARTAMENTO  MUNICIPIO  ESTABLECIMIENTO  DIRECCION  \
SECTOR                                                                          
0.0        992       978           992        992              992        974   
1.0        322       303           322        322              322        320   
2.0      20038     19198         20038      20038            20032      19944   
3.0      11992     11181         11992      11992            11992      11825   

        TELEFONO  SUPERVISOR  DIRECTOR  NIVEL   AREA  STATUS  MODALIDAD  \
SECTOR                                                                    
0.0          848         978       593    992    992     992        992   
1.0          204         303       207    322    322     322        322   
2.0        13362       19197     11190  20038  20038   20038      20038   
3.0        10236       11210      8550  11992  11992   11992      11992   

        JORNADA   PLAN  DEPARTAMENTAL  
SECTOR                                 
0.0         992    992            992  
1.0         322    322            322  
2.0       20038  20038          20038  
3.0       11992  11992          11992

#### Limpiamos la variable "PLAN"

In [20]:
df.groupby("PLAN").count()

CODIGO  DISTRITO  DEPARTAMENTO  MUNICIPIO  ESTABLECIMIENTO  \
PLAN                                                                          
A DISTANCIA         287       167           287        287              286   
DIARIO(REGULAR)   29683     28162         29683      29683            29678   
DOMINICAL            44        40            44         44               44   
FIN DE SEMANA      3223      3198          3223       3223             3223   
INTERCALADO           2         2             2          2                2   
IRREGULAR             2         2             2          2                2   
MIXTO                 4         4             4          4                4   
SABATINO             68        54            68         68               68   
SEMIPRESENCIAL       31        31            31         31               31   

                 DIRECCION  TELEFONO  SUPERVISOR  DIRECTOR  NIVEL  SECTOR  \
PLAN                                                                        
A DISTANCIA            213       112         167       119    287     287   
DIARIO(REGULAR)      29494     21405       28185     17649  29683   29683   
DOMINICAL               43        32          41        24     44      44   
FIN DE SEMANA         3208      3017        3199      2692   3223    3223   
INTERCALADO              2         2           2         2      2       2   
IRREGULAR                2         0           2         2      2       2   
MIXTO                    4         3           4         3      4       4   
SABATINO                66        48          57        18     68      68   
SEMIPRESENCIAL          31        31          31        31     31      31   

                  AREA  STATUS  MODALIDAD  JORNADA  DEPARTAMENTAL  
PLAN                                                               
A DISTANCIA        287     287        287      287            287  
DIARIO(REGULAR)  29683   29683      29683    29683          29683  
DOMINICAL           44      44         44       44             44  
FIN DE SEMANA     3223    3223       3223     3223           3223  
INTERCALADO          2       2          2        2              2  
IRREGULAR            2       2          2        2              2  
MIXTO                4       4          4        4              4  
SABATINO            68      68         68       68             68  
SEMIPRESENCIAL      31      31         31       31             31

In [21]:
categoria_num(df, "PLAN", "DIARIO(REGULAR)", "SABATINO","FIN DE SEMANA", "DOMINICAL", "MIXTO", "A DISTANCIA",
               "INTERCALADO", "IRREGULAR", "SEMIPRESENCIAL")

DIARIO(REGULAR) : 0
 SABATINO : 1
 FIN DE SEMANA : 2
 DOMINICAL : 3
 MIXTO : 4
 A DISTANCIA : 5
 INTERCALADO : 6
 IRREGULAR : 7
 SEMIPRESENCIAL : 8



In [22]:
df.groupby("PLAN").count()

CODIGO  DISTRITO  DEPARTAMENTO  MUNICIPIO  ESTABLECIMIENTO  DIRECCION  \
PLAN                                                                          
0.0    29683     28162         29683      29683            29678      29494   
1.0       68        54            68         68               68         66   
2.0     3223      3198          3223       3223             3223       3208   
3.0       44        40            44         44               44         43   
4.0        4         4             4          4                4          4   
5.0      287       167           287        287              286        213   
6.0        2         2             2          2                2          2   
7.0        2         2             2          2                2          2   
8.0       31        31            31         31               31         31   

      TELEFONO  SUPERVISOR  DIRECTOR  NIVEL  SECTOR   AREA  STATUS  MODALIDAD  \
PLAN                                                                            
0.0      21405       28185     17649  29683   29683  29683   29683      29683   
1.0         48          57        18     68      68     68      68         68   
2.0       3017        3199      2692   3223    3223   3223    3223       3223   
3.0         32          41        24     44      44     44      44         44   
4.0          3           4         3      4       4      4       4          4   
5.0        112         167       119    287     287    287     287        287   
6.0          2           2         2      2       2      2       2          2   
7.0          0           2         2      2       2      2       2          2   
8.0         31          31        31     31      31     31      31         31   

      JORNADA  DEPARTAMENTAL  
PLAN                          
0.0     29683          29683  
1.0        68             68  
2.0      3223           3223  
3.0        44             44  
4.0         4              4  
5.0       287            287  
6.0         2              2  
7.0         2              2  
8.0        31             31

#### Limpiamos la variable "NIVEL"

In [23]:
df.groupby("NIVEL").count()

CODIGO  DISTRITO  DEPARTAMENTO  MUNICIPIO  \
NIVEL                                                             
BASICO                  4948      4777          4948       4948   
DIVERSIFICADO           7970      7728          7970       7970   
PARVULOS                7400      7154          7400       7400   
PREPRIMARIA BILINGUE    2253      2035          2253       2253   
PRIMARIA               10512      9785         10512      10512   
PRIMARIA DE ADULTOS      261       181           261        261   

                      ESTABLECIMIENTO  DIRECCION  TELEFONO  SUPERVISOR  \
NIVEL                                                                    
BASICO                           4947       4861      4096        4777   
DIVERSIFICADO                    7970       7915      7226        7757   
PARVULOS                         7397       7363      4718        7154   
PREPRIMARIA BILINGUE             2253       2247      1455        2035   
PRIMARIA                        10510      10445      7019        9784   
PRIMARIA DE ADULTOS               261        232       136         181   

                      DIRECTOR  SECTOR   AREA  STATUS  MODALIDAD  JORNADA  \
NIVEL                                                                       
BASICO                    3576    4948   4948    4948       4948     4948   
DIVERSIFICADO             6316    7970   7970    7970       7970     7970   
PARVULOS                  4132    7400   7400    7400       7400     7400   
PREPRIMARIA BILINGUE      1442    2253   2253    2253       2253     2253   
PRIMARIA                  4966   10512  10512   10512      10512    10512   
PRIMARIA DE ADULTOS        108     261    261     261        261      261   

                       PLAN  DEPARTAMENTAL  
NIVEL                                       
BASICO                 4948           4948  
DIVERSIFICADO          7970           7970  
PARVULOS               7400           7400  
PREPRIMARIA BILINGUE   2253           2253  
PRIMARIA              10512          10512  
PRIMARIA DE ADULTOS     261            261

In [24]:
categoria_num(df, "NIVEL", "BASICO", "DIVERSIFICADO","PARVULOS", "PREPRIMARIA BILINGUE", "PRIMARIA", "PRIMARIA DE ADULTOS",
               "", "", "")

BASICO : 0
 DIVERSIFICADO : 1
 PARVULOS : 2
 PREPRIMARIA BILINGUE : 3
 PRIMARIA : 4
 PRIMARIA DE ADULTOS : 5
  : 6
  : 7
  : 8



In [25]:
df.groupby("NIVEL").count()

CODIGO  DISTRITO  DEPARTAMENTO  MUNICIPIO  ESTABLECIMIENTO  DIRECCION  \
NIVEL                                                                          
0.0      4948      4777          4948       4948             4947       4861   
1.0      7970      7728          7970       7970             7970       7915   
2.0      7400      7154          7400       7400             7397       7363   
3.0      2253      2035          2253       2253             2253       2247   
4.0     10512      9785         10512      10512            10510      10445   
5.0       261       181           261        261              261        232   

       TELEFONO  SUPERVISOR  DIRECTOR  SECTOR   AREA  STATUS  MODALIDAD  \
NIVEL                                                                     
0.0        4096        4777      3576    4948   4948    4948       4948   
1.0        7226        7757      6316    7970   7970    7970       7970   
2.0        4718        7154      4132    7400   7400    7400       7400   
3.0        1455        2035      1442    2253   2253    2253       2253   
4.0        7019        9784      4966   10512  10512   10512      10512   
5.0         136         181       108     261    261     261        261   

       JORNADA   PLAN  DEPARTAMENTAL  
NIVEL                                 
0.0       4948   4948           4948  
1.0       7970   7970           7970  
2.0       7400   7400           7400  
3.0       2253   2253           2253  
4.0      10512  10512          10512  
5.0        261    261            261

### Pandas Profiling

- se genera un reporte con estadísticas y categorización de los datos.

In [ ]:
#profile = ProfileReport(df, title = "Reporte Proyecto", explorative = True) #generamos el Codebook, le ponemos un titulo y que sea interactivo

In [ ]:
#profile.to_file("Codebook.html") #imprimimos el HTML 

### Link a Github

https://github.com/karinaValladares/CC3066-Proyecto-1